### PR TITLE
chore: improve comments and add tests to trapezoidal rule

### PR DESCRIPTION
**Pull request by jurichar**
### Describe your change:

Contributes to https://github.com/TheAlgorithms/Python/issues/9943
And convert value explicity to float

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: maths/trapezoidal_rule.py [hunk 1]
```diff
@@ -1,47 +1,103 @@
...
 
-method 1:
-"extended trapezoidal rule"
+Method 1:
+"Extended Trapezoidal Rule"
 
 """
 
 
 def method_1(boundary, steps):
```
Comment (tianyizheng02) [inline maths/trapezoidal_rule.py:12]:
```suggestion
def trapezoidal_rule(boundary, steps):
```
Let's rename the function to something more descriptive. I'm not sure why the code calls it "method 1" when there's only one method in the file.

File: maths/trapezoidal_rule.py [hunk 2]
```diff
@@ -1,47 +1,103 @@
 """
 Numerical integration or quadrature for a smooth function f with known values at x_i
 
-This method is the classical approach of suming 'Equally Spaced Abscissas'
+This method is the classical approach of summing 'Equally Spaced Abscissas'
```
Comment (tianyizheng02) [inline maths/trapezoidal_rule.py:4]:
```suggestion
The trapezoidal rule is the classical approach of summing 'Equally Spaced Abscissas'
```
Let's rephrase the header docstring to name the rule directly. I'm not sure why the original author called it "method 1".

File: maths/trapezoidal_rule.py [hunk 3]
```diff
@@ -1,47 +1,103 @@
...
 
-method 1:
-"extended trapezoidal rule"
+Method 1:
+"Extended Trapezoidal Rule"
```
Comment (tianyizheng02) [inline maths/trapezoidal_rule.py:8]:
```suggestion
```
See previous comment

File: maths/trapezoidal_rule.py [hunk 4]
```diff
@@ -1,47 +1,103 @@
...
+    0.0
+
+    >>> f(1)
+    1.0
+
+    >>> f(0.5)
+    0.25
+    """
+    y = float((x - 0) * (x - 0))
     return y
```
Comment (tianyizheng02) [inline maths/trapezoidal_rule.py:83]:
```suggestion
    return x**2
```
Doesn't this do the same thing?

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Apart from my other comments, this file also needs type hints

File: maths/trapezoidal_rule.py (hunk 4 above)
Comment (jurichar) [reply maths/trapezoidal_rule.py:80]: Yes

**Review by jurichar — COMMENTED** [review]
The functions seems tested and goodly commented

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**